homeassistant/components/media_player/sonos.py

```diff
@@ -61,6 +61,10 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     global DEVICES
 
     if discovery_info:
+        # if device allready maped by config
+        if DEVICES:
```

Comment:
If you have two Sonos players and do not use the config entry, it will now only be able to discover 1 entry and will never discover the second one because of this return.

It would be better if you can check if there is already a similar device in `DEVICES`